README.md

```diff
@@ -21,128 +19,504 @@ composer require php-llm/llm-chain
+
+$messages = new MessageBag(Message::ofUser('What date and time is it?'));
+$response = $chain->call($messages, ['response_format' => [
+    'type' => 'json_schema',
+    'json_schema' => [
+        'name' => 'clock',
+        'strict' => true,
+        'schema' => [
+            'type' => 'object',
+            'properties' => [
+                'date' => ['type' => 'string', 'description' => 'The current date in the format YYYY-MM-DD.'],
+                'time' => ['type' => 'string', 'description' => 'The current time in the format HH:MM:SS.'],
+            ],
+            'required' => ['date', 'time'],
+            'additionalProperties' => false,
+        ],
+    ],
+]]);
+
+dump($response->getContent()); // returns an array
+```
+
+#### Code Examples
+
+1. **Structured Output** (PHP class): [structured-output-math.php](examples/structured-output-math.php)
+1. **Structured Output** (array): [structured-output-clock.php](examples/structured-output-clock.php)
+
+### Tool Parameters
+
+LLM Chain generates a JSON Schema representation for all tools in the `ToolBox` based on the `#[AsTool]` attribute and
+method arguments and doc block. Additionally, JSON Schema support validation rules, which are partially support by
+LLMs like GPT.
+
+To leverage this, configure the `#[ToolParameter]` attribute on the method arguments of your tool:
+```php
+use PhpLlm\LlmChain\ToolBox\Attribute\AsTool;
+use PhpLlm\LlmChain\ToolBox\Attribute\ToolParameter;
+
+#[AsTool('my_tool', 'Just an example tool with parameters requirements.')]
+final class MyTool
+{
+    /**
+     * @param string $name   The name of an object
+     * @param int    $number The number of an object
+     */
+    public function __invoke(
+        #[ToolParameter(pattern: '/([a-z0-1]){5}/')]
+        string $name,
+        #[ToolParameter(minimum: 0, maximum: 10)]   
+        int $number,
+    ): string {
+        // ...
+    }
+}
+```
+
+Please be aware, that this is only converted in a JSON Schema for the LLM to respect, but not validated by LLM Chain.
+
+### Response Streaming
+
+Since LLMs usually generate a response word by word, most of them also support streaming the response using Server Side
+Events. LLM Chain supports that by abstracting the conversion and returning a Generator as content of the response.
+
+```php
+use PhpLlm\LlmChain\Chain;
+use PhpLlm\LlmChain\Message\Message;
+use PhpLlm\LlmChain\Message\MessageBag;
+
+// Initialize Platform and LLM
+
+$chain = new Chain($llm);
+$messages = new MessageBag(
+    Message::forSystem('You are a thoughtful philosopher.'),
+    Message::ofUser('What is the purpose of an ant?'),
+);
+$response = $chain->call($messages, [
+    'stream' => true, // enable streaming of response text
+]);
+
+foreach ($response->getContent() as $word) {
+    echo $word;
+}
+```
+
+In a terminal application this generator can be used directly, but with a web app an additional layer like [Mercure](https://mercure.rocks)
```

Comment:
@silasjoisten we can test that, as we have Mercure up and running now

Reply:
Yes